src/inmanta/env.py

```diff
@@ -825,14 +835,31 @@ def get_constraint_violations_for_check(
         """
         Return the constraint violations that exist in this venv. Returns a tuple of non-strict and strict violations,
         in that order.
+
+
         """
+
+        class OwnedRequirement(NamedTuple):
+            requirement: Requirement
+            owner: Optional[str] = None
+
+            def is_owned_by(self, owners: abc.Set[str]) -> bool:
+                return self.owner is None or self.owner in owners
+
         # all requirements of all packages installed in this environment
-        installed_constraints: abc.Set[Requirement] = frozenset(
-            requirement for dist_info in pkg_resources.working_set for requirement in dist_info.requires()
+        installed_constraints: abc.Set[OwnedRequirement] = frozenset(
+            OwnedRequirement(requirement, dist_info.key)
+            for dist_info in pkg_resources.working_set
+            for requirement in dist_info.requires()
         )
-        inmanta_constraints: abc.Set[Requirement] = frozenset(cls._get_requirements_on_inmanta_package())
-        extra_constraints: abc.Set[Requirement] = frozenset(constraints if constraints is not None else [])
-        all_constraints: abc.Set[Requirement] = installed_constraints | inmanta_constraints | extra_constraints
+        inmanta_constraints: abc.Set[OwnedRequirement] = frozenset(
+            OwnedRequirement(r, owner="inmanta-core") for r in cls._get_requirements_on_inmanta_package()
+        )
+        extra_constraints: abc.Set[OwnedRequirement] = frozenset(
+            (OwnedRequirement(r) for r in constraints) if constraints is not None else []
+        )
+
+        all_constraints: abc.Set[OwnedRequirement] = installed_constraints | inmanta_constraints | extra_constraints
```

Comment:
I think it is now possible that this set contains a given requirement multiple times with different owners (which is what we want) or with a null owner.
If such a requirement is present both with a proper owner and a null owner, it will create duplicate VersionConflict (differing only by owner) in the constraint_violations_strict set.
1. I don't know if this is likely to arise
2. I'm not sure whether it's an issue or not

Reply:
I think we should strive to get rid of the null owners over time, but I see your point